src/source/PeerConnection/SessionDescription.c

```diff
@@ -398,7 +398,17 @@ STATUS populateSingleMediaSection(PKvsPeerConnection pKvsPeerConnection, PKvsRtp
     SPRINTF(pSdpMediaDescription->sdpAttributes[attributeCount].attributeValue, "%d", mediaSectionId);
     attributeCount++;
 
-    STRCPY(pSdpMediaDescription->sdpAttributes[attributeCount].attributeName, "sendrecv");
+    // Set appropriate directions in the SDP for the local transceivers
+    if(pKvsRtpTransceiver->transceiver.direction == RTC_RTP_TRANSCEIVER_DIRECTION_SENDRECV) {
+        STRCPY(pSdpMediaDescription->sdpAttributes[attributeCount].attributeName, "sendrecv");
+    }
+    else if(pKvsRtpTransceiver->transceiver.direction == RTC_RTP_TRANSCEIVER_DIRECTION_SENDONLY) {
+        STRCPY(pSdpMediaDescription->sdpAttributes[attributeCount].attributeName, "sendonly");
+    }
+    else if(pKvsRtpTransceiver->transceiver.direction == RTC_RTP_TRANSCEIVER_DIRECTION_RECVONLY) {
+        STRCPY(pSdpMediaDescription->sdpAttributes[attributeCount].attributeName, "recvonly");
+    }
+
```

Comment:
Once you've done the switch/case, you might want to err out perhaps or ignore the case when it's neither.

Reply:
Done